app/controllers/businessTradingName/CheckYourAnswersController.scala

```diff
@@ -123,13 +126,31 @@ class CheckYourAnswersController @Inject() (val errorHandler: ErrorHandler,
 
   def updateBusinessName(): Action[AnyContent] = (authPredicate andThen businessNameAccessPredicate).async { implicit user =>
     if (appConfig.features.businessNameR19_R20Enabled()) {
+
+      val currentBusinessName: Option[String] = user.session.get(validationBusinessNameKey) match {
+        case Some("") => None
+        case Some(name) => Some(name)
+        case _ => None
+      }
+
       user.session.get(prepopulationBusinessNameKey) match {
         case Some(businessName) =>
 
           val updatedBusinessName = UpdateBusinessName(businessName, capacitorEmail = user.session.get(verifiedAgentEmail))
 
           vatSubscriptionService.updateBusinessName(user.vrn, updatedBusinessName) map {
-            case Right(_) => Redirect(controllers.routes.ChangeSuccessController.businessName())
+            case Right(successModel) =>
+              auditingService.audit(ChangedBusinessNameAuditModel(
```

Comment:
The idea behind us being asked to audit the ETMP response status and message is because they wanted us to audit even on submission failures (so, for each of the cases here). Apologies if that wasn't clear on the story. Could you also do the same for the Trading Name equivalent above.